Core/Repositories/SFRepository/GameSFRepository.cs

```
@@ -74,8 +77,8 @@ public async Task<GameDto> FindByIdAsync(Guid id)
                 
             if (gameDto == null)
             {
-                _logger.Error($"Failed to find serialised GameDto file with id {id}");
-                return await Task.FromException<GameDto>(new ArgumentException("Unable to locate a game with provided id"));
+                _logger.LogError($"Failed to find serialised GameDto file with id {id}");
+                throw new HttpStatusCodeException(StatusCodes.Status400BadRequest, @"Failed to find serialised GameDto file with id {id}");

```

You should avoid throwing HTTP related exceptions from repositories (and avoid throwing them from services too), or you will be coupling these to the API layer.

If you want to reuse these repositories/services in other kind of applications that don't use HTTP (e.g. a mobile client, a desktop app, etc.) you shouldn't need to know anything about HTTP.

In a nutshell, the layers usually have an order where API > Services > Repositories. A layer should not reference an upper layer (e.g. repositories should not use services or API, services should not use API). You could have cross-cutting layers where you place infrastructure utils or things that are used by all the layers such as exception type; but even in this case I'd avoid using specific HTTP naming here.

---

(Not referencing upper layers will help to reuse code and avoid some unnecesary headaches related to packages, circular references, etc.).

---

Yeah that definitely makes sense. Would a possible solution to this be create my own exceptions that indicate the type of error that has occured and throw these within the core project. Then in the api middleware I could populate the http response depending on the type of exception that has been thrown?

---

Yep, that is similar to how we do it on Dice if I'm not wrong 👍

---

Yeah I think that's how we do it on dice. Basically you want to throw an exception in the domain and then let the middleware handle how we treat it on the API. If we decided to reuse these repositories on say a Desktop app, you could catch the domain exception and then handled it on the Desktop app (show an error screen for example). If we kept the HTTP exceptions it'd be much harder to integrate.